addon/Feeds/SearchFeed.js

```diff
@@ -23,6 +25,38 @@ module.exports = class SearchFeed extends Feed {
     this.store.dispatch({type: "SEARCH_STATE_UPDATED", data: state});
   }
 
+  /**
+   * getSuggestions - Retrieves search suggestions for a search string (action.data)
+   *                  Dispatches an array of suggestions.
+   */
+  getSuggestions(action) {
+    const browser = getCurrentBrowser();
+    return this.options.searchProvider.asyncGetSuggestions(browser, action.data)
+      .then(suggestions => this.options.send(am.actions.Response("SEARCH_SUGGESTIONS_RESPONSE", suggestions), action.workerId, true))
+      .catch(e => Cu.reportError(e));
```

Comment:
The promise is actually just returned for testing, it doesn't actually get used by anything else